### DISCUSSION TITLE
Position a Menu to the right of Chart

### DISCUSSION THREAD

**Discussion by somguyth** [chakra-ui/chakra-ui#4468] [Q&A] [answered]
I'm adding a `chakra-ui` menu dropdown button to the top-right of a chart (from `react-financial-charts`, which is a library built over `svg`). 

I've partially succeeded by adding a `foreignObject` with `width={100%}` and `height={100%}`, and using a style override to force the button to the top-right.

[![enter image description here][1]][1]

```
<foreignObject  
width={'100%'} height={'100%'} style={{overflow: 'auto'}}>
 <Menu>
    <MenuButton as={Button} rightIcon={<ChevronDownIcon />}         
    transition="all 0.001s"
    borderRadius="md"
    borderWidth="0px"
    _hover={{ bg: "gray.400" }}
    _expanded={{ bg: "blue.400" }}
    _focus={{ boxShadow: "none" }}
    style={{ marginTop: "1px", position: "absolute", right: 0 }}>
      Actions
    </MenuButton>
    <Portal>
    <MenuList zIndex={10}>
      <MenuItem>Download</MenuItem>
      <MenuItem>Create a Copy</MenuItem>
      <MenuItem>Mark as Draft</MenuItem>
      <MenuItem>Delete</MenuItem>
      <MenuItem>Attend a Workshop</MenuItem>
    </MenuList>
    </Portal>
  </Menu>
</foreignObject>
```
However, the problem is that this `foreignObject` occupies **the entire chart** (due to width & height of 100%), and so I can't click/interact with any other part of the chart.

How can I fix this such that I can have the button in the top right without disabling interaction with the rest of the chart?

Codesandbox:

https://codesandbox.io/s/floral-water-v11gx?file=/src/BasicLineSeries.tsx

  [1]: https://i.stack.imgur.com/hy9Af.png

**Comment by somguyth** [comment, marked as answer]
I solved this by constructing `foreignObject` like this:

```
<foreignObject  width={'1'} height={'1'} style={{overflow: 'visible'}} x={this.props.width} y={1}>
```

**Comment by 827142763** [comment]
您好，我已收到您的邮件，谢谢

**Comment by rashid893** [comment]
Who are you
